Firmware/Marlin_main.cpp

```diff
@@ -9321,12 +8618,7 @@ void clamp_to_software_endstops(float target[3])
 
     // Clamp the Z coordinate.
     if (min_software_endstops) {
-        float negative_z_offset = 0;
-        #ifdef ENABLE_AUTO_BED_LEVELING
-            if (Z_PROBE_OFFSET_FROM_EXTRUDER < 0) negative_z_offset = negative_z_offset + Z_PROBE_OFFSET_FROM_EXTRUDER;
-            if (cs.add_homing[Z_AXIS] < 0) negative_z_offset = negative_z_offset + cs.add_homing[Z_AXIS];
-        #endif
-        if (target[Z_AXIS] < min_pos[Z_AXIS]+negative_z_offset) target[Z_AXIS] = min_pos[Z_AXIS]+negative_z_offset;
+        if (target[Z_AXIS] < min_pos[Z_AXIS]) target[Z_AXIS] = min_pos[Z_AXIS];
```

Comment:
If you're wondering where 14 bytes of flash are saved. It is here, by removing `negative_z_offset` since it is always 0.